cartridge/roles/coordinator.lua

```diff
@@ -298,34 +313,47 @@ local function appoint_leaders(leaders)
 
         local replicaset = replicasets[k]
         if replicaset == nil then
-            return nil, AppointmentError:new('Replicaset "%s" does not exist', k)
+            return nil, AppointmentError:new(
+                "Replicaset %q doesn't exist", k
+            )
         end
 
         local server = servers[v]
         if server == nil then
-            return nil, AppointmentError:new('Server "%s" does not exist', v)
+            return nil, AppointmentError:new(
+                "Server %q doesn't exist", v
+            )
         end
 
         if server.replicaset_uuid ~= k then
-            return nil, AppointmentError:new('Server "%s" does not belong to replicaset "%s"', v, k)
+            return nil, AppointmentError:new(
+                "Server %q doesn't belong to replicaset %q", v, k
+            )
         end
+    end
+
+    local session = vars.session
```

Comment:
If you don't want to force reconnect client if session is broken/closed/... add "client.get_session_or_nil" function or smth else with better name.